drivers/azure/azure.go

```diff
@@ -525,8 +558,10 @@ func (d *Driver) Remove() error {
 	if err := c.DeleteNetworkSecurityGroupIfExists(ctx, d.nsgResource, d.nsgUsedInPool); err != nil {
 		return err
 	}
-	if err := c.CleanupAvailabilitySetIfExists(ctx, d.ResourceGroup, d.AvailabilitySet); err != nil {
-		return err
+	if d.AvailabilityZone == "" {
```

Comment:
Could be helpful to add a comment on _why_ we are cleaning up any existing Availability Set if we see the Availability Zone is empty here